File: go-gin/rest/issueDto.go
package rest

import (
	"errors"
	"lizzy/medium/compare/go-gin/domain"

	"github.com/gin-gonic/gin"
	"github.com/google/uuid"
)

type issueDto struct {
	Id          uuid.UUID
	Name        string
	Description string
}

func parseBody(c *gin.Context) (issueDto, error) {
	var issueDto issueDto
	if err := c.ShouldBindJSON(&issueDto); err != nil {
		return issueDto, &gin.Error{
			Err:  errors.New("Cannot parse body: " + err.Error()),
			Type: gin.ErrorTypePublic}
	}
	return issueDto, nil
}

func issueToIssueDto(issue domain.Issue) issueDto {
	return issueDto{issue.GetId(), issue.GetName(), issue.GetDescription()}
}

func (i issueDto) toIssue() domain.Issue {
	return domain.NewIssue(i.Id, i.Name, i.Description)
}

File: node-ts/src/rest/issueDto.ts
import {Issue} from '../domain/issue'
import { isUuid } from 'uuidv4';
import { InvalidIdException } from './issueController';

export class IssueDto {
    public id : string = "";
    public name : string = "";
    public description: string = "";

    static fromIssue(issue : Issue) : IssueDto{
        const issueDto = new IssueDto();
        issueDto.id = issue.getId();
        issueDto.name = issue.getName();
        issueDto.description = issue.getDescription();
        return issueDto;
    }

    static fromJson(json: any): Issue {
        const issueDto = new IssueDto();
        console.log(json)
		if(!isUuid(json.id)){
			throw new InvalidIdException(json.id);
		}
        return new Issue(json.id, json.name, json.description);
    }
}